frontend/catalyst/jax_extras/tracing.py

```diff
@@ -536,11 +544,29 @@ def make_jaxpr_f(*args, **kwargs):
 
 
 def input_type_to_tracers(
-    in_type: InputType, maker: Callable[[AbstractValue], DynamicJaxprTracer]
+    in_type: InputType,
+    arg_maker: Callable[[AbstractValue], DynamicJaxprTracer],
+    const_maker: Callable[[DynamicJaxprTracer], DynamicJaxprTracer],
 ) -> List[DynamicJaxprTracer]:
     """Creates an expanded list of tracers representing an input values of a Jaxpr program"""
-    in_aval, _ = unzip2(in_type)
-    return _input_type_to_tracers(maker, in_aval)
+    in_tracers: list[DynamicJaxprTracer] = []
+
+    def _substitute_tracers_in_aval(a: AbstractValue) -> AbstractValue:
```

Comment:
Is there not an existing function doing that in Jax? What is the difference wwith unzip2?

Reply:
I found the following implementation of [unzip2](https://github.com/google/jax/blob/main/jax/_src/util.py)

```python
def unzip2(xys: Iterable[tuple[T1, T2]]
    ) -> tuple[tuple[T1, ...], tuple[T2, ...]]:
  """Unzip sequence of length-2 tuples into two tuples."""
  # Note: we deliberately don't use zip(*xys) because it is lazily evaluated,
  # is too permissive about inputs, and does not guarantee a length-2 output.
  xs: list[T1] = []
  ys: list[T2] = []
  for x, y in xys:
    xs.append(x)
    ys.append(y)
  return tuple(xs), tuple(ys)
```

This is different. The "expansion" is obtaining the list of parameters from all sources. As a note, in JAX with dynamic shapes, types can have a parameter. E.g., tensor of size "A". So the expansion iterates through the parameter types and adds them as to the list of input tracers.